File: CodeGolf/Thor/Thor.py
# Readable version

a,b,c,d=map(int,input().split())
while 1:
    x,y=(c<a)-(a<c),(d<b)-(b<d)
    print((" SN"[y]+" EW"[x]).strip())
    c+=x
    d+=y

# Code golf

a,b,c,d=map(int,input().split())
while 1:x,y=(c<a)-(a<c),(d<b)-(b<d);print((" SN"[y]+" EW"[x]).strip());c+=x;d+=y

File: CodeGolf/Thor/Thor.js
// Readable version

[x,y,X,Y]=readline().split(' ')
for(;;){
    print((y>Y++?"S":y<--Y?(Y--,"N"):"") + (x>X++?"E":x<--X?(X--,"W"):""))
}

// Code Golf

[x,y,X,Y]=readline().split(' ');for(;;){print((y>Y++?"S":y<--Y?(Y--,"N"):"")+(x>X++?"E":x<--X?(X--,"W"):""))}